zendriver/core/config.py

```diff
@@ -78,11 +78,11 @@ def __init__(
         if not browser_args:
             browser_args = []
 
-        if not user_data_dir:
-            self._user_data_dir = temp_profile_dir()
-            self._custom_data_dir = False
-        else:
-            self.user_data_dir = user_data_dir
+        # defer creating a temp user data dir until the browser requests it so
+        # config can be used/reused as a template for multiple browser instances
+        self._user_data_dir: Optional[str] = None
+        if user_data_dir:
+            self.user_data_dir = str(user_data_dir)
```

Comment:
I see the setter handles setting this to True, but I don't think we have anything to set it to False in case `Config.uses_custom_data_dir` is called before `Config.user_data_dir`

```suggestion
        if user_data_dir:
            self.user_data_dir = str(user_data_dir)
        else:
            self._custom_data_dir = False
```

Reply:
I just submitted another commit after testing again to be sure. I've added code to make sure all cases are caught and it passes `mypy` review as well. I can squash the commits if you would like after review.

Reply:
Great, thank you.

No need to squash, I'll just use the automatic GitHub squash feature